packages/compartment-mapper/src/import-hook.js

```diff
@@ -68,10 +91,67 @@ const nodejsConventionSearchSuffixes = [
   '/index.node',
 ];
 
+/**
+ * Given a module specifier which is an absolute path, attempt to match it with
+ * an existing compartment; return a {@link RedirectStaticModuleInterface} if found.
+ *
+ * @throws If we determine `absoluteModuleSpecifier` is unknown
+ * @param {FindRedirectParams} params Parameters
+ * @returns {RedirectStaticModuleInterface|undefined} A redirect or nothing
+ */
+const findRedirect = ({
+  compartmentDescriptor,
+  compartmentDescriptors,
+  compartments,
+  absoluteModuleSpecifier,
+  packageLocation,
+}) => {
+  let specifierIsArbitrary = true;
+  for (const [someDescriptorName, someDescriptor] of entries(
+    compartmentDescriptors,
+  )) {
+    if (someDescriptorName !== ATTENUATORS_COMPARTMENT) {
+      const moduleSpecifierUrl = `${new URL(absoluteModuleSpecifier, packageLocation)}`;
+      if (!compartmentDescriptor.location.startsWith(moduleSpecifierUrl)) {
+        if (moduleSpecifierUrl.startsWith(someDescriptor.location)) {
+          // compartmentDescriptor is a dependency of someDescriptor; we
+          // can use that compartment
+          if (someDescriptor.name in compartmentDescriptor.modules) {
+            return {
+              specifier: absoluteModuleSpecifier,
+              compartment: compartments[someDescriptorName],
+            };
+          }
+          // compartmentDescriptor is a dependent of someSomeDescriptor;
+          // this is more dubious but a common pattern
+          if (compartmentDescriptor.name in someDescriptor.modules) {
```

Comment:
I don't understand why this is a case we want to handle.

If the absolute path import matches a compartment (someDescriptor) it makes sense to check if curernt compartment descriptor has that in the collection of modules, but why would we check the other way around?

Reply:
Maybe the thought process above would be easier to do if we added a comment explaining why the checks are there.

Also, the ifs can be merged.